allennlp/training/model.py

```diff
@@ -13,9 +13,9 @@ class Model(torch.nn.Module):
     data generation, because model inputs are named.
     """
 
-    def forward(self,
+    def forward(self,  # pylint: disable=arguments-differ
                 inputs: Dict[str, torch.Tensor],
-                compute_loss: bool = False):  # pylint: disable=arguments-differ
+                compute_loss: bool = False) -> (Union[torch.Tensor, List[torch.Tensor]], Optional[float]):
```

Comment:
oh right and change the data api to return inputs and targets dictionaries separately or something? At the moment we don't have a distinction between them, which would make doing this hard without guessing at the names people give targets etc. Sounds good to me.